akka-stream/src/main/scala/akka/stream/StreamTimeoutException.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+import scala.concurrent.TimeoutException
+import scala.util.control.NoStackTrace
+import akka.annotation.DoNotInherit
+
+/**
+ * Base class for timeout exceptions specific to Akka Streams
+ *
+ * Not for user extension
+ */
+@DoNotInherit
+sealed class StreamTimeoutException(msg: String) extends TimeoutException(msg) with NoStackTrace
+
+final class InitialTimeoutException(msg: String) extends StreamTimeoutException(msg)
+
+final class CompletionTimeoutException(msg: String) extends StreamTimeoutException(msg)
+
+final class StreamIdleTimeoutException(msg: String) extends StreamTimeoutException(msg)
```

Comment:
Why is this prefixed with Stream? Others are not. I think we can skip Stream here.

Reply:
There is already an TcpIdleTimeoutException , should this be StreamIdleTimeoutException?

Reply:
this was already discussed https://github.com/akka/akka/pull/31640#pullrequestreview-1130828577